Automattic/newspack-ios
# Issues/58 data folder reconciliation

Refs #58 

This PR makes improvements to the way the site's folder is managed. 
- Site folder creation logic is improved and moved into the SiteStore
- Tests are added to check folder naming rules.
- Tests are modified to resolve some inconsistent test results after the SiteStore changes.
- A reference to the sites's folder is now saved in core data. 
- Logic is added to recreate a missing folder when the SiteStore is created (at the start of a session) or when the app is resumed.
- Some things were renamed for consistency. 

Notes:
- The logic to recreate a deleted folder when the app is resumed is rudimentary and will be improved as individual story folders (and folder contents) are dealt with--it works to test the concept for now but order of folder reconstruction (or clean up of core data relations) will be an issue.
- I'm also reconsidering the role of the FolderManager's currentFolder. It seemed like a good tool to scope new folder/asset creation but it needs to be reset which is an annoying implementation detail. I'll think more about this and deal with it another time.
- Finally, tests seem cranky lately, something related to async save operations I think. Locally they pass 9/10. CI might need to be kicked to sort them there.

To test:
- Confirm all tests pass. 
- Delete the app from the simulator or device so there is a fresh start.
- Install and launch the app via this branch.
- Confirm browse to the folder list, then tap back.
- Switch to the files app. Check that a site folder was created correctly.
- Select the folder in the Files app and delete it. 
- Resume the Newspack app and confirm that you can browse to the folder list. (The list will be empty if you resumed. This is expected for now.)
- Return to the files app and confirm that the folder was recreated.

@jleandroperez Can I trouble you for a review?  This one is easy to test but there's more going on code wise.
